cson/cson.c

```diff
@@ -518,18 +528,18 @@ int cson_array_append(cson_array *ar, cson_value *v)
     cson_value_free(v);
     return 0;
 }
-int cson_array_reserve(cson_array *ar, unsigned int size)
-{
-    return 0;
-}
 cson_value *cson_value_new_integer(cson_int_t v)
 {
     cson_value *val = calloc(1, sizeof(cson_value));
     val->sql_type = SQLITE_INTEGER;
     val->u.num = v;
-    char buf[128];
-    val->value_bytes = sprintf(buf, "%" PRId64, v);
-    val->value_text = strndup(buf, val->value_bytes);
+    val->value_bytes = snprintf(val->value_buf, sizeof(val->value_buf), "%" PRId64, v);
+    if (val->value_bytes < sizeof(val->value_buf)) {
+        val->value_text = val->value_buf;
+    } else {
+        val->value_text = malloc(val->value_bytes + 1);
+        sprintf(val->value_text, "%" PRId64, v);
```

Comment:
not sure why you cant do here: 
  val->value_text = strndup(buf, val->value_buf);
but if you have to malloc, instead of sprintf you can do:
  strcpy(val->value_text, val->value_buf);

Reply:
I can't do that because `value_buf` is too small to hold stringy-fied integer.

Reply:
This and other such codepaths are taken when stringy-fied versions of value is larger than 48 bytes. So most small strings, double values, integer values etc never hit this codepath.

Reply:
I understand, maybe we could keep the original char buf[128]; version, int count = sprintf() into it, and if (count < 48) { strcpy(value_buf, buf) } else {value_buf = strdup(buf)};

Reply:
I think it is better to optimize for common case where payload fits in buffer and no additional copies or allocations are done. Still, I've increased size of buf to 128 to catch even larger things (fingerprints, small queries, etc). Let is review this if it is actually a performance issue.